### DISCUSSION TITLE
CORS - 405 Error

### DISCUSSION THREAD

**Discussion by go4cas** [nitrojs/nitro#2155] [Q&A] [answered]
Are there any known issues with preflight (OPTIONS) in a Nitro standalone (without Nuxt) server? I'm getting "405 HTTP method is not allowed." My `nitro.config.js`:
```javascript
export default defineNitroConfig({
  srcDir: 'src/',
  rootDir: '.',
  runtimeConfig: {
    googleClientId: '',
    jwtSecret: '',
    jwtStrategy: '',
  },
  alias: {
    '@': '~',
  },
  routeRules: {
    '/auth/**': {
      cors: true,
      headers: { 'access-control-allow-methods': '*', 'access-control-allow-credentials': 'true', 'access-control-allow-origin': 'http://localhost:8080' },
    },
  },
})
```

**Comment by go4cas** [comment]
I got this working by removing the `routeRules` completely from `nitro.config.js`, and adding the following to the top of the route handler (as per suggestion in discussion https://github.com/unjs/nitro/discussions/1677#discussioncomment-6863355):
```javascript
export default defineEventHandler(async (event) => {
  handleCors(event, {
    origin: ['http://localhost:8080'],
    credentials: true,
    methods: '*',
    allowHeaders: '*',
  })
  if (event.node.req.method === 'OPTIONS') {
    return null
  }
  const body = await readBody(event)
  ...
  ...
  ...
}
```

Although this is working as expected now, I am not sure what the use case for the `cors` option in the `routeRules` config is?

**Reply by jamaluddinrumi** [reply, marked as answer]
you can put 

```typescript
handleCors(event, {
  origin: ['http://localhost:8080'],
  credentials: true,
  methods: '*',
  allowHeaders: '*',
})
if (event.node.req.method === 'OPTIONS') {
  return null
}
```

in middleware, so you don't have to put that cors-related task in every your routes.
